src/python/dxpy/bindings/dxdataobject_functions.py

```diff
@@ -146,16 +146,32 @@ def get_handler(id_or_link, project=None):
 
 def describe(id_or_link, **kwargs):
     '''
-    :param id_or_link: String containing an object ID or dict containing a DXLink
+    :param id_or_link: String containing an object ID or dict containing a DXLink,
+                       or a list of object ID's or dict's containing a DXLink.
 
     Given an object ID, calls :meth:`~dxpy.bindings.DXDataObject.describe` on the object.
 
     Example::
 
         describe("file-1234")
     '''
-    handler = get_handler(id_or_link)
-    return handler.describe(**kwargs)
+    # If this is a list, extract the ids.
+    if isinstance(id_or_link, list):
+        links = []
+        for link in id_or_link:
+            # If this entry is a dxlink, then get the id.
+            if is_dxlink(link):
+                # Guaranteed by is_dxlink that one of the following will work
+                if isinstance(link['$dnanexus_link'], basestring):
+                    link = link['$dnanexus_link']
+                else:
+                    link = link['$dnanexus_link']['id']
+            links.append(link)
+        retval = dxpy.api.system_describe_data_objects({'objects': links})
```

Comment:
Call this something other than `retval` since it's not the value that you're eventually going to return from this method.